controllers/jobController.js

```diff
@@ -74,50 +34,90 @@ module.exports = {
             company_name: req.body.company_name,
             description: req.body.description,
         };
-
-        Job.findByIdAndUpdate(jobId, {
-            $set: jobParams
-        })
-            .then(job => {
-                let user = res.locals.user;
-                req.flash('success', `The job offer has been updated successfully!`);
-                res.redirect(`/user/${user._id}/offers`);
-                next();
-            })
-            .catch(error => {
+    //  Job.findByIdAndUpdate(jobId, { $set: jobParams})
+        Job.findOneAndUpdate({_id: jobId}, {$set: jobParams}, {new: true}, (err, job) => {
+            if(err) {
                 req.flash('error', `There has been an error while updating the job offer: ${error.message}`);
                 console.log(`Error updating job by ID: ${error.message}`);
-                next(error);
-            });
-    },
-
+            } else {
+                //let user = res.locals.user;
+                req.flash('success', `The job has been successfully updated!`);
+                res.redirect(`/user/${user._id}/offers`);
+                console.log('job updated in MongoDB and Elasticsearch')
+            }
+        });
+     },
+     
     /**
      * Delete the job offer from the whole jobs collection &
-     * delete this job offer from the collection of job offers 
-     * (jobOffers array in User) create by particular user.
+     * from the collection of job offers (jobOffers array in User) create by particular user.
      */
     deleteJob: (req, res) => {
         let jobId = req.params.jobId;
+        let user = req.user;
 
-        Job.findByIdAndDelete(jobId)
-            .then((job) => {
-                let user = res.locals.user;
-                console.log('user to update:', user);
-                User.findByIdAndUpdate(user._id, {
-                    $pull: {
```

Comment:
this line is important, we can not delete it. because the job offer we delete is deleted from job offers collections, but still stays in the user's array. That is why we need to use **pull** to delete it also from the user.

Reply:
Screenshots: 
1. the job offer is still in the user.
![image](https://user-images.githubusercontent.com/38938392/102072420-28c83c00-3e02-11eb-8f16-d34ca10c6693.png)
2.  despite the fact that job offer is deleted from jobs collection.
![image](https://user-images.githubusercontent.com/38938392/102072455-341b6780-3e02-11eb-96c2-2083698a3417.png)

Reply:
thank youuu for pointing this out! i must've been really tired but i didn't think about this